tmt/steps/provision/connect.py

```diff
@@ -110,8 +111,31 @@ def go(self):
             data['key'] = key
 
         # And finally create the guest
-        self._guest = tmt.Guest(data, name=self.name, parent=self.step)
+        self._guest = GuestConnect(data, name=self.name, parent=self.step)
 
     def guest(self):
         """ Return the provisioned guest """
         return self._guest
+
+
+class GuestConnect(tmt.Guest):
+    """
+    Guest connection class.
+    """
+
+    def reboot(self, hard=False, machine_name=None):
+        super.reboot(hard, name)
```

Comment:
```suggestion
        super().reboot(hard, machine_name)
```
This would throw an exception - you must call `super` and also `name` is undefined.

Reply:
Thanks. Solved.